components/compliance-service/inspec-agent/resolver/resolver.go

```diff
@@ -259,6 +259,70 @@ func handleRegionFilters(filters []*common.Filter, regions []string) ([]string,
 }
 
 func (r *Resolver) handleAwsApiNodes(ctx context.Context, m *manager.NodeManager, filters []*common.Filter, job *jobs.Job) ([]*types.InspecJob, error) {
+	// we have upgraded our logic to be able to work with mult-region scanning, meaning any new integrations
+	// created will only create one node per account (instead of one per region)
+	// the following logic checks to see if more than one node is associated with the account in our db
+	// if it is, we follow the "old path"
+	// if it is not, we follow a "new path" where we just prepare the one node for scanning
```

Comment:
I'm a little confused about how this works. The old method creates entries in the nodes table if they don't exist, but the new method assumes the one node is already there. Where does that node get created?

Reply:
Does this need to handle the case when 0 nodes exist?

Reply:
Ok, I think I just needed to read further. It looks like the nodes get created in `/nodemanager-service/api/manager/server/server.go`. That means the "old path" could probably be simplified but it's probably worth keeping it as is because its in a known working state.

Reply:
Also we expect the things to always be in the database, so 0 results here should probably be an error.

Reply:
good point - i'll push an update to return an error if there are 0 results here